gcovr/timestamps.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding:utf-8 -*-
+
+#  ************************** Copyrights and license ***************************
+#
+# This file is part of gcovr 5.0, a parsing and reporting tool for gcov.
+# https://gcovr.com/en/stable
+#
+# _____________________________________________________________________________
+#
+# Copyright (c) 2021 the gcovr authors
+# This software is distributed under the BSD License.
+# For more information, see the README.rst file.
+#
+# ****************************************************************************
+
+import datetime
+import re
+
+UTC = datetime.timezone.utc
+
+
+def parse_timestamp(value: str) -> datetime.datetime:
+    r"""
+    Parse a timestamp.
+
+    This can either use use "YYYY-MM-DD hh:mm:ss" or epoch notation.
+
+    For future compatibility, the timestamp may start
+    with an alphanumeric prefix (think: URL-scheme)
+    that forces a particular parsing.
+
+    Implemented prefixes (case-insensitive):
+
+    * `epoch:` or `@` to parse the number as an Unix epoch
+    * `rfc3339:` to get the "ISO" format, with the following changes:
+      A timezone offset is optional.
+      The "T" separator may be replaced by a space.
+
+    Examples of parsing epochs:
+
+    >>> parse_timestamp("1640606727").isoformat()  # no prefix
+    '2021-12-27T12:05:27+00:00'
+    >>> parse_timestamp("@1640606727").isoformat()
+    '2021-12-27T12:05:27+00:00'
+    >>> parse_timestamp("epoch:1640606727").isoformat()
+    '2021-12-27T12:05:27+00:00'
+
+    Examples of parsing RFC 3339 like timestamps:
+
+    >>> parse_timestamp("2021-12-27 13:05:27").isoformat()  # no prefix
+    '2021-12-27T13:05:27'
+    >>> parse_timestamp("rfc3339:2021-12-27 13:05:27").isoformat()
```

Comment:
```suggestion
    >>> parse_timestamp("rfc3339:2021-12-27T3:05:27").isoformat()
```
To also test the default separator.